ext/standard/file.c

```diff
@@ -344,7 +344,9 @@ PHP_FUNCTION(flock)
 
 	PHP_STREAM_TO_ZVAL(stream, res);
 
-	act = operation & 3;
+	act = operation & PHP_LOCK_UN;
+	// TODO doesn't this fail if operation is a bitmask with LOCK_NB?
```

Comment:
Assuming that `LOCK_NB==4` and `PHP_LOCK_UN==3`, the former would have already been masked out.

Reply:
I'm not very good at understanding masks so I assume that you're correct